app/controllers/Admin.scala

```diff
@@ -207,5 +212,87 @@ object Admin extends Controller with PanDomainAuthActions {
     )
   }
 
+  /*
+    Section Tag association
+   */
+
+  val addSectionTagForm = Form(
+    mapping(
+      "section_id" -> longNumber,
+      "tag_id" -> nonEmptyText
+    )(assignTagToSectionFormData.apply)(assignTagToSectionFormData.unapply)
+  )
+
+  val removeSectionTagForm = Form(
+    mapping(
+      "section_id" -> longNumber,
+      "tag_id" -> nonEmptyText
+    )(unAssignTagToSectionFormData.apply)(unAssignTagToSectionFormData.unapply)
+  )
+
+  def sectionsAndTags(selectedSectionIdOption: Option[Long]) = (AuthAction andThen WhiteListAuthFilter).async {
+
+    val selectedSectionOptionFt:Future[Option[Section]] = selectedSectionIdOption match {
+      case Some(selectedSectionId) => {
+        for {
+          sections <- getSortedSections()
+        }yield{
+          Some(sections.filter( x => x.id == selectedSectionId ).head)
+        }
+      }
+      case None => Future(None)
+    }
+    val tagIdsOptionFuture: Future[Option[List[String]]] = selectedSectionIdOption match {
```

Comment:
> the list of tags should always be a list 

... which is going to be empty if either
1. No section was selected ( in which case  `selectedSectionId` is `None`), or
2. Was selected but there was no tag against it. 

Cheers,